### PR TITLE
Add some typing to ``NamedTuple`` brain

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] Write a good description on what the PR does.

## Description

This extracts some changes from #1306 to make that PR more reviewable.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Related Issue

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM. Maybe something for 2.11.0 ?

**Comment by DanielNoord** [comment]
> LGTM. Maybe something for 2.11.0 ?

No rush here. It needs to be merged before #1306 can land, but we can put this in `2.10.1` or `2.11` depending on when this gets reviewed by Marc.

File: astroid/brain/brain_namedtuple_enum.py [hunk 1]
```diff
@@ -90,7 +92,12 @@ def _extract_namedtuple_arg_or_keyword(  # pylint: disable=inconsistent-return-s
     raise UseInferenceDefault()
 
 
-def infer_func_form(node, base_type, context=None, enum=False):
+def infer_func_form(
+    node: nodes.Call,
+    base_type: List[nodes.NodeNG],
```
Comment (cdce8p) [inline astroid/brain/brain_namedtuple_enum.py:97]:
```suggestion
    base_type: nodes.NodeNG,
```

Comment (DanielNoord) [reply astroid/brain/brain_namedtuple_enum.py:97]: Shit sorry.. Thought I double checked this 😓

**Review by cdce8p — APPROVED** [review]

**Pull request merged**